client/middleware/card-transitions.ts

```diff
@@ -0,0 +1,26 @@
+const VALID_ROUTES = ['login', 'register', 'reset-password']
+
+const SLIDE_LEFT = { name: 'slide-left' }
+const SLIDE_RIGHT = { name: 'slide-right' }
+
+const INDICES = new Map<string, number>([
+  ['register', -2],
+  ['reset-password', -1],
+  ['login', 0],
+])
+
+export default defineNuxtRouteMiddleware((to, from) => {
+  if (!VALID_ROUTES.includes(to.name)) return
+  if (!VALID_ROUTES.includes(from.name)) return
+
+  const toIndex = INDICES.get(to.name)!
+  const fromIndex = INDICES.get(from.name)!
+
```

Comment:
(along with the other comment about the Map)

```suggestion
  const toIndex = INDICES[to.name as keyof typeof INDICES]
  const fromIndex = INDICES[from.name as keyof typeof INDICES]
```